test/common/pixel-tests

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -eu
+
+SUBDIR=test/reference
+STORAGE_REPO=../pixel-test-reference
+
+message() {
+    [ "${V-}" != 0 ] || printf "  %-8s %s\n" "$1" "$2"
+}
+
+cmd_init() {
+    git submodule add -b empty "$STORAGE_REPO" test/reference
+}
+
+cmd_pull() {
+    git submodule update --init -- "$SUBDIR"
+}
+
+cmd_status() {
+    ( cd "$SUBDIR"
+      git rm --cached --quiet '*.png'
+      git add *.png
+      if git diff-index --name-status --cached --exit-code HEAD; then
+          echo No changes
+      fi
+    )
+}
+
+cmd_push() {
+    ( cd "$SUBDIR"
+      git rm --cached --quiet '*.png'
+      git add *.png
+      if ! git diff-index --name-status --cached --exit-code HEAD; then
+          git checkout --quiet --orphan temp
+          git commit --quiet -m "$(date)"
+          head=$(git rev-parse HEAD)
+          git -c advice.detachedHead=false checkout --quiet "$head"
+          git branch --quiet -d temp
+          git tag "sha-$head" HEAD
+          git push origin "sha-$head"
```

Comment:
I still don't understand what's the deal with this -- can you please add a comment? If this is merely about having independent "branch-less" tags on the github repo? Then it should be enough to just push the tag, but not the branch (i.e. exactly what this git push already does). Or is it really important to *not* commit this to the local checkout/repo?

Reply:
It's about having commits without parent.  But commits with the "empty"` branch as their parent are just as good, and might be much easier to produce.

Reply:
Done.